File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int n=nums.size();
        
        for(int i=0;i<n-1;i++)
        {
            if(nums[i] == 0)
            {
                int j = i+1;
                while(nums[j]==0 && j<n-1)
                    j++;
                swap(nums[i],nums[j]);
            }
        }
        
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    
    public void moveZeroes(int[] nums) 
    {
        int n = nums.length,nZ=-1,k=0;
        for(int i=0;i<n;i++)
        {
            if(nums[i]!=0) {if(nZ==-1) nZ = i;k++;}
        }
        if(nZ == -1) return ;
        
        int i=0;
        while(i<k)
        {
            nums[i] = nums[nZ];
            nZ++;
            while(nZ<n && nums[nZ]==0) nZ++;
            i++;
        }
        for(i=k;i<n;i++) nums[i]=0;
    }
}